src/daft-local-execution/src/sinks/cross_join_collect.rs

```diff
@@ -0,0 +1,124 @@
+use std::sync::{Arc, OnceLock};
+
+use common_error::DaftResult;
+use common_runtime::RuntimeRef;
+use daft_micropartition::MicroPartition;
+use daft_table::Table;
+
+use super::blocking_sink::{
+    BlockingSink, BlockingSinkFinalizeResult, BlockingSinkSinkResult, BlockingSinkState,
+    BlockingSinkStatus,
+};
+
+pub(crate) type CrossJoinStateBridgeRef = Arc<CrossJoinStateBridge>;
```

Comment:
why do we need a custom bridge here? We're just sending over a micropartition. Our standard channel / oneshot should work right? @colin-ho

Reply:
This micropartition needs to be broadcasted to all workers, but the standard channel currently does not do that.

Reply:
if that is the case, let's rework this to a more generic broadcast bridge. We should just be able to make this a generic.
`BroadcastStateBridge<T>` and use it for both Probe Table and `Micropartition`

Reply:
Note: Not blocking this PR for this ^^

Reply:
Will add a todo for colin then